### PR TITLE
Create Eval Plan document .docx template

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
# Overview
Added new template and path in `generate-document.ts` to accept an `EvaluationPlan` and return a `.docx`

### CODE REVIEW

File: document-generation/generate-document.ts
Comment (jasonburkert-ccpo) [inline document-generation/generate-document.ts:44]: Control should flow through the switch/case statement until it hits a break or return statement.
I would expect something like this rather than introducing a logical or in the case value :
```suggestion
    case DocumentType.INCREMENTAL_FUNDING_PLAN:
    case DocumentType.EVALUATION_PLAN:
    // return generateWordDocument(event); -- satisfies both of the above
```

File: document-generation/eval-plan-document.test.ts
Comment (juliusfitzhugh-ccpo) [inline document-generation/eval-plan-document.test.ts:14]: Not sure if it was intentional or not, but I might want to update the template path.

File: document-generation/eval-plan-document.ts
Comment (juliusfitzhugh-ccpo) [inline document-generation/eval-plan-document.ts:16]: Do you plan on updating other values inside the payload? If not then you probably don't need to spread the payload here.

File: document-generation/generate-document.ts
Comment (juliusfitzhugh-ccpo) [inline document-generation/generate-document.ts:47]: I think these cases can be consolidate sense the same function is being called. And I started using `Docx`, so there might be a small merge conflict for the naming depending on when things are merged.

```typescript

    case DocumentType.INCREMENTAL_FUNDING_PLAN:
    case DocumentType.EVALUATION_PLAN:
      return generateWordDocument(event);
```

File: models/document-generation.ts
Comment (juliusfitzhugh-ccpo) [inline models/document-generation.ts:589]: Just a heads up if notice on the other document schemas the `additionalProperties: false` is used. Make sure to include it because it helps with the usage of `oneOf` used by the main `generateDocumentSchema` otherwise it won't pass validation.

File: document-generation/utils/utils.ts
Comment (juliusfitzhugh-ccpo) [inline document-generation/utils/utils.ts:13]: Is this being used for something?

**Review by juliusfitzhugh-ccpo — COMMENTED** [review]
I know its still in draft, but I did  a quick look over and left a few minor comments.

File: document-generation/generate-document.ts
Comment (zachclark-ccpo) [reply document-generation/generate-document.ts]: Implemented in https://github.com/dod-ccpo/atat-web-api/pull/1381/commits/59e999694dcb21b3f0c4bca134d475dd4bb68221

File: document-generation/eval-plan-document.ts
Comment (zachclark-ccpo) [reply document-generation/eval-plan-document.ts:19]: Might be worth it to keep the payload this way if there are future changes.

File: document-generation/eval-plan-document.test.ts
Comment (zachclark-ccpo) [reply document-generation/eval-plan-document.test.ts]: This was unintentional, fixed in https://github.com/dod-ccpo/atat-web-api/pull/1381/commits/619ea2ac7b0ace6681d64e26cc2d39871dd631d9

**Review by juliusfitzhugh-ccpo — DISMISSED** [review]
Overall looks good. A few minor things in the code. Also, nice job the `IF` blocks are working nicely in the template! I did notice some things in the Eval Plan .docx template (see below), but I'm no expert so you might want to double check with @luakim-ccpo.

### Code
- update HOTH spec and JSON schema to have `taskOrderTitle`
- might want to account for a `null` value in the `method` property I think you can add 
```typescript
enum: [..., EvalPlanMethod.LPTA, null],
```
### Eval Plan Template
For the conditions in the `IF` blocks be careful with the type of quote used and make sure it's not the curly one that Word auto changes (I'm not sure if my document auto corrected them when I opened it or not). The curly quote caused issues when trying to generate the doc. Using the back tick "(`)"  is another option, which is what I used in another template.

#### Additional fields to populate
- Include `standard differentiators` in `BVTO` section, since I think all but the last `optional` line is a predefined text selection in SNOW records (see `Eval Plan Differentiator` table records)
- Include `standard specifications` in `Best Use` section,  2 of the `optional` are in SNOW (see `Eval Plan Assessment Area` table records)
- Include `standard specifications` in `Lowest Risk` section,  1 of the `optional` is in SNOW
- Also for the last `EQUAL_SET_LUMP_SUM` option the `method` is `null` I think

File: models/document-generation.ts
Comment (juliusfitzhugh-ccpo) [reply models/document-generation.ts:603]: This suggestion is unrelated to the question directly above this, but happens to be in the same area related to the mention of the `method` being `null` above
```suggestion
    method: {
      type: "string",
      enum: [EvalPlanMethod.BEST_USE, EvalPlanMethod.LOWEST_RISK, EvalPlanMethod.BVTO, EvalPlanMethod.LPTA, null],
    },
```

Comment (zachclark-ccpo) [reply models/document-generation.ts]: Added in https://github.com/dod-ccpo/atat-web-api/pull/1381/commits/b12f4330c5046458ab4e7bbb3e3dd0a8c194e41c

File: models/document-generation.ts
Comment (zachclark-ccpo) [reply models/document-generation.ts:593]: Added in https://github.com/dod-ccpo/atat-web-api/pull/1381/commits/ee30dba17ddfca8dc84c925f376aea2e0f5bb596

File: models/document-generation.ts
Comment (zachclark-ccpo) [reply models/document-generation.ts:296]: Updated in https://github.com/dod-ccpo/atat-web-api/pull/1381/commits/dc255916ea84ac4649455cc0c4ca2433b033a436

**Comment by jasonburkert-ccpo** [comment]
> ### Eval Plan Template
> For the conditions in the `IF` blocks be careful with the type of quote used and make sure it's not the curly one that Word auto changes (I'm not sure if my document auto corrected them when I opened it or not). The curly quote caused issues when trying to generate the doc. Using the back tick "(`)" is another option, which is what I used in another template.

@juliusfitzhugh-ccpo Do you advise using back tick character to avoid the curly quotes issue altogether?

**Comment by zachclark-ccpo** [comment]
> > ### Eval Plan Template
> > For the conditions in the `IF` blocks be careful with the type of quote used and make sure it's not the curly one that Word auto changes (I'm not sure if my document auto corrected them when I opened it or not). The curly quote caused issues when trying to generate the doc. Using the back tick "(`)" is another option, which is what I used in another template.
> 
> @juliusfitzhugh-ccpo Do you advise using back tick character to avoid the curly quotes issue altogether?

I have had any issues generating the document locally or using the deployed AWS sandbox branch endpoint, but I will add backticks as a failsafe.

According to JC, template should be good (no required `standard differentiators`, etc) (check Teams discussion).

**Comment by juliusfitzhugh-ccpo** [comment]
> @juliusfitzhugh-ccpo Do you advise using back tick character to avoid the curly quotes issue altogether?

Yes. It might have been only my computer changing the curly quotes (on some but not on all 🤷 ) and caused minor issues when trying to test on sandbox stack.

File: models/document-generation.ts
Comment (juliusfitzhugh-ccpo) [inline models/document-generation.ts:603]: Because of the addition of `null` the `type: "string"` will cause the request to fail validation if the method is `null`.

```suggestion
      enum: [EvalPlanMethod.BEST_USE, EvalPlanMethod.LOWEST_RISK, EvalPlanMethod.BVTO, EvalPlanMethod.LPTA, null],
```

**Review by juliusfitzhugh-ccpo — COMMENTED** [review]
Updates to the code and template look good! Also, I like the color coding for the `IF`s blocks. Aside from getting @luakim-ccpo approval on each variation before merging, should be good to go.

**Review by juliusfitzhugh-ccpo — CHANGES_REQUESTED** [review]
Changes to `Eval Plan` template needed based on feedback.

**Comment by zachclark-ccpo** [comment]
Just got it approved.

**Review by juliusfitzhugh-ccpo — APPROVED** [review]

**Pull request merged**
